djconnectwise/sync.py

```diff
@@ -3148,6 +3161,24 @@ def _assign_field_data(self, instance, json_data):
             instance.record_type = models.Ticket.PROJECT_TICKET
 
         self.set_relations(instance, json_data)
+
+        # Check if project is None or has closed status - skip if so
+        if instance.project is None:
+            raise InvalidObjectException(
+                'Project ticket {} has no project - skipping.'.format(
+                    instance.id
+                )
+            )
+
+        # Check if the project has a closed status
+        if (instance.project.status and
+            instance.project.status.closed_flag):
```

Comment:
You don't need to check for this, if the project is in the DB then the ticket stays, if the project isn't in the db, then it is dropped.